packages/ember-glimmer/lib/component-managers/custom.ts

```diff
@@ -0,0 +1,101 @@
+import { assert } from 'ember-debug';
+import AbstractManager from './abstract';
+import {
+  CurlyComponentLayoutCompiler
+} from './curly';
+import { ComponentManager } from '@glimmer/runtime';
+import ComponentStateBucket from '../utils/curly-component-state-bucket';
+import { ComponentCapabilities } from './capabilities';
+
+/**
+  Wrapper class for custom component managers as per
+  Custom Components RFC.
+
+  When defining a custom component manager, the following
+  methods are required:
+
+  - `create`
+  - `getSelf`
+  - `update`
+
+  `layoutFor` method is not required to be implemented and
+  mirrors the behaviour of curly components.
+  @private
+  @class CustomManager
+*/
+export default class CustomManager extends AbstractManager<ComponentStateBucket> {
+  private _manager: ComponentManager<ComponentStateBucket>;
+
+  constructor(manager) {
+    super();
+
+    this._manager = manager;
+
+    assert('You must implement `create` method.', this._manager.create !== undefined);
+    assert('You must implement `getSelf` method.', this._manager.getSelf !== undefined);
+    assert('You must implement `update` method.', this._manager.update !== undefined);
+  }
+
+  create(environment, definition, args, dynamicScope, callerSelfRef, hasBlock) {
+    return this._manager.create(environment, definition, args, dynamicScope, callerSelfRef, hasBlock);
+  }
+
+  // state is ComponentDefinition
+  getCapabilities(state): ComponentCapabilities {
+    // should likely return default?
+    return state.capabilities;
+  }
+
+  getSelf({ component }) {
+    return this._manager.getSelf(component);
+  }
+
+  update(bucket, dynamicScope) {
+    return this._manager.update(bucket, dynamicScope);
+  }
+
+  layoutFor(definition, bucket, env) {
+    if (this._manager.layoutFor) {
+      return this._manager.layoutFor(definition, bucket, env);
```

Comment:
I'm not 100% of the args here.  I generally think that all of the manager hooks should receive the state bucket as their first argument, but the thing I am _really_ unsure about is if we should be exposing the `env` or `definition`'s here.  It _somewhat_ infers that they would become public API which I don't think is what we want.